sdk/messaging/azservicebus/sender.go

```diff
@@ -93,7 +106,13 @@ type SendMessageBatchOptions struct {
 // Message batches can be created using [Sender.NewMessageBatch].
 // If the operation fails it can return an [*azservicebus.Error] type if the failure is actionable.
 func (s *Sender) SendMessageBatch(ctx context.Context, batch *MessageBatch, options *SendMessageBatchOptions) error {
-	err := s.links.Retry(ctx, EventSender, "SendMessageBatch", func(ctx context.Context, lwid *internal.LinksWithID, args *utils.RetryFnArgs) error {
+	var err error
+	ctx, endSpan := s.startSpan(ctx, "SendMessageBatch", tracing.SendOperationName,
+		tracing.Attribute{Key: tracing.BatchMessageCount, Value: int64(batch.NumMessages())},
+	)
+	defer func() { endSpan(err) }()
+
+	err = s.links.Retry(ctx, EventSender, "SendMessageBatch", func(ctx context.Context, lwid *internal.LinksWithID, args *utils.RetryFnArgs) error {
 		return lwid.Sender.Send(ctx, batch.toAMQPMessage(), nil)
```

Comment:
@lmolkova, in cases like this where I have retries, should the span reporting be _in_ the retry loop, or outside, like we have here?

Reply:
The way it works in our HTTP SDKs is that the method span is above the retry layer, and its child HTTP span is below the retry layer. So you'd have spans like this.
```
Some method call span
  HTTP span retry 1
  HTTP span retry 2
```
I presume we'd want to do the same thing here.

Reply:
@jhendrixMSFT I did a bit of digging -- the semantic conventions for HTTP has examples for how the spans should look like for retries: https://opentelemetry.io/docs/specs/semconv/http/http-spans/#http-client-authorization-retry-examples. However, for messaging systems there is not much information besides [this chunk](https://opentelemetry.io/docs/specs/semconv/messaging/messaging-spans/): 

![image](https://github.com/user-attachments/assets/71f50f2d-4159-4778-bfd0-e02d4434bab8)

I am not sure if this implies the messaging spans are 1 per operation?